### ISSUE TITLE
Warping in dense pointcloud

### ISSUE THREAD

**Issue by eriksobolt** [cdcseacave/openMVS#688]
**Describe the bug**
Warping in result of densification step. I am using the combination of colmap and openmvs to recunstruct facades. I have experienced some warping in the dense point cloud which propagates through the whole pipeline. Usually these are areas where the overlap is pretty low, which can probably expected. However at some points I would expect this to go a bit better.
Furthermore, colmap seems to handle this a bit better by default. Are there any settings which I could tweak to come to a better result?

I would like to keep using this awesome project, since I need the high resolution that the textured result produces

The colmap screenshot is from a textured mesh. So I have no problems with the holes in the dense point cloud, these will be filled during the texturing step. It is the warping only

**openmvs**
![Screenshot from 2021-07-30 10-06-46](https://user-images.githubusercontent.com/81301541/127622988-01f93141-891e-480a-b944-de3f67d3ae20.png)

**colmap**
![image](https://user-images.githubusercontent.com/81301541/127623169-5bf3ee33-5115-4de6-ade5-8af2eb2e8920.png)

**Comment by cdcseacave** [comment]
I do not understand what you mean by `warping`, but I imagine the problem are the holes. The holes can occur for few reasons, like textureless regions, poor overlap, bad image quality (variable/bad illumination/exposure, etc). Trying to be more robust in textureless regions you can increase the patch size during PatchMatch: `nSizeHalfWindow` in `DepthMap.h`

**Comment by eriksobolt** [comment]
sorry, the holes are not the problem. You see that the bricks and the mortar have no straight lines (this is what i meant with warping). In colmap these are reconstructed correctly (sorry for the low image quality)

**OpenMVS warped**
![image](https://user-images.githubusercontent.com/81301541/127630433-7b81249d-b3e2-4377-9f90-ea5eecebe519.png)

**Colmap**

![image](https://user-images.githubusercontent.com/81301541/127630657-109abce4-c00b-4e17-8c17-0f6aa3a2dda7.png)

**Comment by cdcseacave** [comment]
pls share the original photos

**Comment by eriksobolt** [comment]
https://drive.google.com/file/d/1tvFo6hRKSWHUcbVR-nKMwh_aag23Lrwy/view?usp=sharing

thanks

**Comment by cdcseacave** [comment]
it reconstructs great on my end, most probably the SfM stage or import is bogus in your case
![dense_map](https://user-images.githubusercontent.com/2802670/127638815-dd7f709e-34a8-4a44-b36a-783d2b3ab690.gif)

**Comment by eriksobolt** [comment]
could you maybe share your script/parameters? The screenshot I shared in the first post was a result from colmap with the same sfm.
Seems like I am doing something pretty wrong somewhere

**Comment by cdcseacave** [comment]
![image](https://user-images.githubusercontent.com/2802670/127642470-59eebca8-80ef-49d5-963b-89173cab7cc4.png)
![image](https://user-images.githubusercontent.com/2802670/127642559-9f428cd9-8b98-4898-9070-472890d712a7.png)

**Comment by cdcseacave** [comment]
I did not use COLMAP for SfM, as I have my own solver, and I am not familiar with COLMAP params, but they look ok. You have to undistort the images and import the linear cameras.

Or you should try OpenMVG

**Comment by cdcseacave** [comment]
![dense_map2](https://user-images.githubusercontent.com/2802670/127643335-a25e3bb0-fb70-42a3-9fd8-77996cd67bf2.gif)

**Comment by eriksobolt** [comment]
Thanks a lot for your time and your help. I got it working, it was indeed something with the undistorted images. In the script I used above I fused the InterfaceVisualSFM. Using the InterfaceColmap gave me the correct result: https://github.com/cdcseacave/openMVS/issues/360 

Really appreciated!

![Screenshot from 2021-07-30 14-11-21](https://user-images.githubusercontent.com/81301541/127651229-2a3249a0-9954-4360-8fc2-28592b68a4e6.png)
![Screenshot from 2021-07-30 14-10-45](https://user-images.githubusercontent.com/81301541/127651232-65401b29-e5ae-4d44-8f1b-5be75b9ea314.png)
